### PR TITLE
Added CRT submodules, updated autoconf

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[submodule "aws-c-common"]
+	path = crt/aws-c-common
+	url = https://github.com/awslabs/aws-c-common.git
+[submodule "aws-c-io"]
+	path = crt/aws-c-io
+	url = https://github.com/awslabs/aws-c-io.git
+[submodule "aws-c-http"]
+	path = crt/aws-c-http
```
Comment (graebm) [inline .gitmodules:8]: nice job giving these the shortest possible path

File: Makefile.frag [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	$(INSTALL_DIR)/lib/libaws-c-compression.a \
+	$(INSTALL_DIR)/lib/libaws-c-http.a \
+	$(INSTALL_DIR)/lib/libaws-c-auth.a \
+	$(INSTALL_DIR)/lib/libs2n.a
+
+crt-static: $(crt_libs) $(INSTALL_DIR)/lib/libcrypto.a
+
+build/deps/openssl:
+	@echo Fetching libcrypto source
+	git clone --single-branch --branch OpenSSL_1_1_1 https://github.com/openssl/openssl.git build/deps/openssl
```
Comment (graebm) [inline Makefile.frag]: trivial: `--depth 1` to speed things up even more?

Comment (graebm) [reply Makefile.frag]: shouldn't the branch be [OpenSSL_1_1_1-stable](https://github.com/openssl/openssl/tree/OpenSSL_1_1_1-stable)?

**Review by graebm — APPROVED** [review]
fix & ship

File: Makefile.frag (hunk 2 above)
Comment (justinboswell) [reply Makefile.frag]: Good catch!

**Pull request merged**